registry/credentials.go

```diff
@@ -4,14 +4,23 @@ import (
 	"encoding/base64"
 	"encoding/json"
 	"fmt"
-	"github.com/pkg/errors"
 	"net/url"
 	"strings"
+
+	"github.com/pkg/errors"
 )
 
 // Registry Credentials
 type creds struct {
-	username, password string
+	username, password   string
+	registry, provenance string
+}
+
+func (c creds) String() string {
+	if c.username == "" && c.password == "" && c.registry == "" && c.provenance == "" {
```

Comment:
This can be `if c == creds{}` and that's better because

Reply:
Oh yeah!